.github/workflows/linux.bash

```diff
@@ -61,22 +63,32 @@ case "$DISTRO" in
         ;;
     esac
 
-    dnf install -y bison ccache cmake gcc-c++ flex ninja-build \
+    dnf install -y bison ccache cmake gcc-c++ flex ninja-build redhat-rpm-config \
       {boost,libedit,mariadb,ncurses,openssl,postgresql,systemd}-devel
     ;;
 esac
 
+case "$DISTRO" in
+  debian:*|ubuntu:*)
+    CMAKE_OPTS+=(-DICINGA2_LTO_BUILD=ON)
+    source <(dpkg-buildflags --export=sh)
+    ;;
+  *)
+    CMAKE_OPTS+=(-DCMAKE_{C,CXX}_FLAGS="$(rpm -E '%{optflags} %{?march_flag}')")
```

Comment:
According to [RPM macros that provide compiler and linker flags](https://docs.fedoraproject.org/en-US/packaging-guidelines/RPMMacros/#_macros_providing_compiler_and_linker_flags), you also need to extract the linker flags from `build_ldflags`.

Reply:
I'm afraid SLES/OpenSuse still need the [`_smp_mflags`](https://git.icinga.com/packages/icinga2/-/blob/main/icinga2.spec.in?ref_type=heads#L460).

Reply:
Is that even suse specific RPM macro?

Reply:
These are only for make(1) which we don't use here.

Reply:
Do you have reference/docs? Isn't that similar to `march_flag `?

Reply:
```
$ docker run --rm -it opensuse/leap
Unable to find image 'opensuse/leap:latest' locally
latest: Pulling from opensuse/leap
5b48f6350ac6: Pull complete
Digest: sha256:8ee016fee97061ba9328cee04bad6fc35462118750a6d7d15b9ab78bc089a247
Status: Downloaded newer image for opensuse/leap:latest
266b694398bb:/ # rpm -E '%{?_smp_mflags}'
-j11
266b694398bb:/ #
```

Reply:
Btw. AL2 and suse pipelines didn't even complain about vector out-of-bound access in https://github.com/Icinga/icinga2/pull/10365, so there's nothing we can do here.